scan/scan_common.go

```diff
@@ -163,57 +170,81 @@ func (fs FamilySet) RunScans(host, ip, family, scanner string, dur time.Duration
 		return nil, err
 	}
 
-	results := make(chan map[string]FamilyResult)
-	timeout := make(chan bool)
+	resultChan := make(chan *Result)
 
-	if dur > 0 {
-		go startTimer(dur, timeout)
-	}
+	var familyWG sync.WaitGroup
+	familyWG.Add(len(fs))
 
-	familyResults := make(map[string]FamilyResult)
+	// Mark as done after completing all scans in all Families
+	go func() {
+		familyWG.Wait()
+		close(resultChan)
+	}()
 
 	go func() {
 		for familyName, family := range fs {
+			var scannerWG sync.WaitGroup
 			if familyRegexp.MatchString(familyName) {
-				scannerResults := make(map[string]ScannerResult)
+				scannerWG.Add(len(family.Scanners))
 				for scannerName, scanner := range family.Scanners {
-					if scannerRegexp.MatchString(scannerName) {
-						grade, output, err := scanner.Scan(addr, hostname)
-						if grade != Skipped {
-							result := ScannerResult{
-								Grade:  grade.String(),
-								Output: output,
+					go func(scannerName string, scanner *Scanner) {
```

Comment:
It's just that there are so many constant globals. It's awkward to have to pass around a `*sync.WaitGroup` and `*regexp.Regexp`.

```Go
func (scanner *Scanner) concurrentScan(familyName, scannerName string, scannerRegexp *regexp.Regexp, scannerWG *sync.WaitGroup) {
...
}
```
```Go
go scanner.concurrentScan(familyName, scannerName, scannerRegexp, scannerWG)
````